### PR TITLE
Dyn 4843 watch node smaller

### PR DESCRIPTION
**Pull request by jesusalvino**
### Purpose

- Set the minimum height to the Watch Node when is created.
- Set the proper heights to the Watch Nodes when a Dynamo file is opened: The minimum height when is a single line, otherwise the Default height.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

###Reviewers
@QilongTang

###FYIs
@RobertGlobant20 @filipeotero

### CODE REVIEW

**Review by QilongTang — COMMENTED** [review]
Looks good on my side. @Amoursol how is the new look?

**Comment by Amoursol** [comment]
> Looks good on my side. @Amoursol how is the new look?

Looking much better for sure! Is it possible to also not have it quite as wide if we are returning a smaller digit like a 2? 

Also, @jesusalvino can you please make a quick GIF of also showcasing the following:

`small = 2;
long = 10000;
list = 0..20;
doubleList = [0..20,20..40];
longSingle = Line.ByStartPointEndPoint(
	Point.ByCoordinates(10,20,30),
	Point.ByCoordinates(40,50,60));`

https://user-images.githubusercontent.com/12857357/165958029-81234e54-6f61-4f5b-af4e-9402623f0804.mov

This will allow us to test a range of cases, and see how the node behaves.

**Comment by jesusalvino** [comment]
> 



> > Looks good on my side. @Amoursol how is the new look?
> 
> Looking much better for sure! Is it possible to also not have it quite as wide if we are returning a smaller digit like a 2?
> 
> Also, @jesusalvino can you please make a quick GIF of also showcasing the following:
> 
> `small = 2; long = 10000; list = 0..20; doubleList = [0..20,20..40]; longSingle = Line.ByStartPointEndPoint( Point.ByCoordinates(10,20,30), Point.ByCoordinates(40,50,60));`
> WatchNodeTesting.mov
> 
> This will allow us to test a range of cases, and see how the node behaves.
WatchSize5

File: src/DynamoCoreWpf/Views/Preview/WatchTree.xaml.cs [hunk 1]
```diff
@@ -26,15 +28,42 @@ public WatchTree()
         void WatchTree_Loaded(object sender, RoutedEventArgs e)
         {
             _vm = this.DataContext as WatchViewModel;
+            _vm.PropertyChanged += _vm_PropertyChanged;
```
[the remark this answers is not in the document]
Comment (jesusalvino) [reply src/DynamoCoreWpf/Views/Preview/WatchTree.xaml.cs:37]: Done

**Comment by reddyashish** [comment]
@jesusalvino @Amoursol What do you think about setting a maximum width and having a horizontal scroll bar. The last case seems pretty wide.

**Review by reddyashish — APPROVED** [review]
Looks good to me once we confirm the max width case.

**Comment by QilongTang** [comment]
Please confirm @Amoursol I also think the last case is too wide to display

**Comment by Amoursol** [comment]
> @jesusalvino @Amoursol @QilongTang What do you think about setting a maximum width and having a horizontal scroll bar. The last case seems pretty wide.

Agreed, it sounded good on paper but looks a bit wild in practice! Can we please go with a width that is reasonable, then the user can expand again? 

Worth testing = 2x the width of the watch node in 2.14 as a max stretch default.

**Comment by jesusalvino** [comment]
WatchSize6

The max width of a Watch node will be its default 2X = 400, if the line exceeds it, the horizontal scrollbar will be available for the user

**Comment by Amoursol** [comment]
> WatchSize6
> 
> @Amoursol The max width of a Watch node will be its default 2X = 400, if the line exceeds it, the horizontal scrollbar will be available for the user

This is looking AWESOME! Thank you @jesusalvino. Did you manage to have it not as wide when returning a small number such as `2` alone?

**Comment by jesusalvino** [comment]
> > WatchSize6
> > @Amoursol The max width of a Watch node will be its default 2X = 400, if the line exceeds it, the horizontal scrollbar will be available for the user
> 
> This is looking AWESOME! Thank you @jesusalvino. Did you manage to have it not as wide when returning a small number such as `2` alone?

imagen

yes

**Comment by Amoursol** [comment]
> > > WatchSize6
> > > @Amoursol The max width of a Watch node will be its default 2X = 400, if the line exceeds it, the horizontal scrollbar will be available for the user
> > 
> > 
> > This is looking AWESOME! Thank you @jesusalvino. Did you manage to have it not as wide when returning a small number such as `2` alone?
> 
> imagen
> 
> yes

Looks perfect to me - thank you! Consider it done from my side 😊

**Review by zeusongit — APPROVED** [review]

**Pull request merged**